File: TransCoder_tokenized_test_set_functions/java/FIND_A_TRIPLET_THAT_SUM_TO_A_GIVEN_VALUE_2.java
static boolean find3Numbers ( int A [ ], int arr_size, int sum ) {
  for ( int i = 0;
  i < arr_size - 2;
  i ++ ) {
    HashSet < Integer > s = new HashSet < Integer > ( );
    int curr_sum = sum - A [ i ];
    for ( int j = i + 1;
    j < arr_size;
    j ++ ) {
      if ( s . contains ( curr_sum - A [ j ] ) && curr_sum - A [ j ] != ( int ) s . toArray ( ) [ s . size ( ) - 1 ] ) {
        System . out . printf ( "Triplet is %d, %d, %d", A [ i ], A [ j ], curr_sum - A [ j ] );
        return true;
      }
      s . add ( A [ j ] );
    }
  }
  return false;
}

File: results/py/FIND_A_TRIPLET_THAT_SUM_TO_A_GIVEN_VALUE_2.cpp
bool f_filled ( int A [ ] , int arrSize , int sum ) {
  for ( int i = 0 ;
  i <= arrSize - 1 ;
  i ++ ) {
    set < int > s ;
    int currSum = sum - A [ i ] ;
    for ( int j = i + 1 ;
    j < arrSize ;
    j ++ ) {
      if ( ( currSum - A [ j ] ) < s . count ( A [ j ] ) ) {
        cout << "Triplet is" << A [ i ] << ", " << A [ j ] << ", " << currSum - A [ j ] << endl ;
        return true ;
      }
      s . insert ( A [ j ] ) ;
    }
  }
  return false ;
}
